File: src/Klc.LogicRoute.Domain/Entities/Shipment.cs
using Klc.LogicRoute.Domain.Common;
using Klc.LogicRoute.Domain.Enums;

namespace Klc.LogicRoute.Domain.Entities;

public class Shipment : BaseEntity
{
    public string ShipmentNumber { get; set; } = string.Empty;
    public Guid? OrderId { get; set; }
    public string? OriginAddress { get; set; }
    public string? OriginCity { get; set; }
    public string? DestinationAddress { get; set; }
    public string? DestinationCity { get; set; }
    public ShipmentStatus Status { get; set; } = ShipmentStatus.Draft;
    public ShipmentPriority Priority { get; set; } = ShipmentPriority.Normal;
    public DateTime? RequestedPickupDate { get; set; }
    public DateTime? RequestedDeliveryDate { get; set; }
    public DateTime? ActualPickupDate { get; set; }
    public DateTime? ActualDeliveryDate { get; set; }

    // Kargo Hesaplama
    public decimal TotalWeightKg { get; set; }
    public decimal TotalVolumeM3 { get; set; }
    public decimal TotalDesiWeight { get; set; }
    public decimal ChargeableWeight { get; set; }
    public int PalletCount { get; set; }
    public bool IsHazardous { get; set; }
    public bool RequiresColdChain { get; set; }
    public decimal? TemperatureMin { get; set; }
    public decimal? TemperatureMax { get; set; }
    public bool IsStackable { get; set; } = true;

    // Karar Motoru Sonucu
    public Guid? SelectedProviderId { get; set; }
    public Guid? SelectedContractRateId { get; set; }
    public VehicleCategory RecommendedVehicle { get; set; }
    public decimal? CalculatedPrice { get; set; }
    public string? Currency { get; set; } = "TRY";
    public string? ProviderReferenceId { get; set; }

    // Takip
    public decimal? CurrentLatitude { get; set; }
    public decimal? CurrentLongitude { get; set; }
    public DateTime? LastTrackingUpdate { get; set; }
    public string? EstimatedArrival { get; set; }

    public string? DriverName { get; set; }
    public string? DriverPhone { get; set; }
    public string? VehiclePlate { get; set; }
    public string? Notes { get; set; }

    public List<ShipmentItem> Items { get; set; } = [];
    public Recommendation? Recommendation { get; set; }
}

File: mobile/composeApp/src/commonMain/kotlin/com/klc/logicroute/data/model/Shipment.kt
package com.klc.logicroute.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Shipment(
    val id: String,
    @SerialName("shipmentNumber")
    val shipmentNumber: String,
    val status: ShipmentStatus,
    @SerialName("originAddress")
    val originAddress: String,
    @SerialName("destinationAddress")
    val destinationAddress: String,
    @SerialName("originCity")
    val originCity: String = "",
    @SerialName("destinationCity")
    val destinationCity: String = "",
    @SerialName("customerName")
    val customerName: String = "",
    @SerialName("customerPhone")
    val customerPhone: String = "",
    @SerialName("weight")
    val weight: Double = 0.0,
    @SerialName("volume")
    val volume: Double = 0.0,
    @SerialName("palletCount")
    val palletCount: Int = 0,
    @SerialName("vehiclePlate")
    val vehiclePlate: String = "",
    @SerialName("driverName")
    val driverName: String = "",
    @SerialName("scheduledDate")
    val scheduledDate: String = "",
    @SerialName("originLat")
    val originLat: Double? = null,
    @SerialName("originLng")
    val originLng: Double? = null,
    @SerialName("destinationLat")
    val destinationLat: Double? = null,
    @SerialName("destinationLng")
    val destinationLng: Double? = null
)
